sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -193,58 +201,53 @@ export class EventHubClient {
    * @param credential - SharedKeyCredential object or your credential that implements the TokenCredential interface.
    * @param options - The options that can be provided during client creation.
    */
-  constructor(
-    host: string,
-    eventHubPath: string,
-    credential: SharedKeyCredential | TokenCredential,
-    options?: EventHubClientOptions
-  );
+  constructor(host: string, eventHubPath: string, credential: TokenCredential, options?: EventHubClientOptions);
   constructor(
     hostOrConnectionString: string,
     eventHubPathOrOptions?: string | EventHubClientOptions,
-    credential?: SharedKeyCredential | TokenCredential,
+    credentialOrOptions?: TokenCredential | EventHubClientOptions,
     options?: EventHubClientOptions
   ) {
     let connectionString;
-    let cred: TokenCredential | SharedKeyCredential;
+    let credential: TokenCredential | SharedKeyCredential;
     hostOrConnectionString = String(hostOrConnectionString);
 
-    if (typeof eventHubPathOrOptions !== "string") {
+    if (!isTokenCredential(credentialOrOptions)) {
       connectionString = hostOrConnectionString;
-      options = eventHubPathOrOptions;
+      if (typeof eventHubPathOrOptions !== "string") {
```

Comment:
If user provides connection string and event hub path, and the connection string contains `EnitytPath`, that should be over-ridden by the explicit event hub path that was provided

I would suggest the below refactoring
```javascript
connectionString = hostOrConnectionString;
const parsedCS = parseConnectionString<EventHubConnectionStringModel>(connectionString);
credential = new SharedKeyCredential(parsedCS.SharedAccessKeyName, parsedCS.SharedAccessKey);
if (typeof eventHubPathOrOptions !== "string") {
   if (!parsedCS.EntityPath) {
     // throw error
   }
   options = eventHubPathOrOptions;
} else {
        parsedCS.EntityPath = eventHubPathOrOptions;
       options = credentialOrOptions;
}

Reply:
This refactoring will not work if user will provide the connectionString (without `EntitytPath`) and `eventHubPath`

Reply:
discussed offline and we found a middle ground